package.json

```diff
@@ -116,7 +116,8 @@
     "compilerOptions": {
       "esModuleInterop": true,
       "lib": [
-        "esnext"
+        "esnext",
+        "DOM"
```

Comment:
this means the types of undici only works if dependent project also adds the dom types. That shouldn't be necessary for a module which explicitly _doesn't_ work in the browser

Reply:
Whether it works in the browser or not is not the relevant part. The types are and fetch is based on DOM types per the spec. So it's not weird that we also need to import DOM types as well...

Reply:
this PR uses `DOMException` (and possibly more) from DOM types

>Whether it works in the browser or not is not the relevant part.

It's relevant for consumers that `undici` doesn't work in a typescript environment that isn't configured to be run in the browser. But I can agree it's more of a symptom than a cause 🙂

Reply:
unfortunately @types/node doesn't seem to include quite a bit. If you remove that line and run `npm run test:typescript` we get errors about missing DOMException and EventInit types.

Reply:
`DOMException` doesn't exist in Node, so that makes sense. `EventInit` seems to be an interface? Not something in MDN at least

Reply:
But just like this project comes with types for most of `fetch` (https://github.com/nodejs/undici/blob/main/types/fetch.d.ts) that are duplicated from TS core dom types, `DOMException` and `EventInit` should also be defined here

Reply:
DOMException exists in core since v17 (somewhere around there) globally, and [EventInit](https://dom.spec.whatwg.org/#dictdef-eventinit) is the name of the options passed to Event.

Reply:
Ah. I tried to run `typeof DOMException` and got undefined, but I see I wrote `typeof DOMEXception` 🙈 

Regardless, I'm sure definetely typed would welcome a PR adding `DOMException`, `EventInit` and any other missing interfaces. But they shouldn't be pulled from `dom`.

EDIT: I do see `EventInit` there already: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/488ef79fe14ea591166813a01ba7f34dfd955b68/types/node/dom-events.d.ts#L78

Reply:
EventInit doesn't seem like a problem regardless, we can create our own and it won't overlap with the existing type (although I'm not sure why I can't access it). If we could "feature detect" DOMException, and only use our own type when it doesn't exist, that would be good. That way, the types won't overlap when/if DOMException gets added to node types.

Reply:
I tried adding `/// <reference types="node/dom-events" />` but the type still couldn't be used. I'll play around a little when I get home.

Reply:
I opened a PR to fix this https://github.com/nodejs/undici/pull/1762

Reply:
I do think this is kind of a TypeScript issue: https://github.com/microsoft/TypeScript/issues/43972